RAG/notebooks/langchain/langchain_copilot_with_NIM_HF_FAISS_deployed_locally.ipynb

```diff
@@ -0,0 +1,510 @@
+   "source": [
+    "## Create Vector Data Base (VDB) using your Proprietary Documents (PART 1)\n",
+    "<div style=\"text-align:center\">\n",
+    "  <img src=\"./data/imgs/offline.png\" alt=\"Alternative text\" />\n",
+    "</div>\n",
+    "\n",
+    "In this section, we will create vector embeddings from your documents and store them in a vector database. We will be following the below-listed steps\n",
+    "\n",
+    "1. Download the embedding model from Huggingface using LangChain plugin\n",
+    "2. Parse all the PDF documents in a folder and break them into text chunks.\n",
+    "3. Pass the chunks to the embedding model to create embeddings.\n",
+    "4. Save the generated embeddings into a FAISS vector database.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f1515cc3-baa5-453a-a002-e3e6f1c5b31d",
+   "metadata": {},
+   "source": [
+    "### Import all the required libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "5b5d433c-e722-46a2-9ec3-891e56120761",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import os\n",
+    "from langchain.document_loaders import PyPDFLoader\n",
+    "from langchain.text_splitter import RecursiveCharacterTextSplitter\n",
+    "from langchain.embeddings import HuggingFaceEmbeddings\n",
+    "from langchain.vectorstores import FAISS\n",
+    "import faiss\n",
+    "import warnings\n",
+    "warnings.filterwarnings('ignore')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8b23788d-87da-4f14-8a9d-4b64699d149e",
+   "metadata": {},
+   "source": [
+    "### Embedding model and global variables\n",
+    "\n",
+    "We provide the name of the embedding model and the location of where the VDB needs to be stored. We also initialize few global variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "997706c9-9095-47b3-8681-e555b55823d4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_name = \"Alibaba-NLP/gte-large-en-v1.5\"\n",
+    "vectorDB_name = \"./papers.db\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "2cc203ea-0afb-41d2-bd57-922181716cca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize global variables\n",
+    "vectorstore = None\n",
+    "embeddings = None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cbf762f0-2173-4aa1-be5c-0b4319f53593",
+   "metadata": {},
+   "source": [
+    "### Download embedding model\n",
+    "\n",
+    "In this function, we download the embedding model from Huggingface using the Huggingface plugin in LangChain. In this function we also check for the availability of GPU/s and pass that as an argument in the `encode` and `model` arguments."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "5fb73eb5-c1fd-4d38-bcf5-53be8d5e062a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def dowload_embedding_model(model_name):\n",
```

Comment:
typo "download"

Reply:
it's consistent across all instances but still a typo

Reply:
corrected.